### PR TITLE
[SEDONA-353] Add RS_BandNoDataValue

### PR DESCRIPTION
**Pull request by iGN5117**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-353. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?
- Add RS_BandNoDataValue

## How was this patch tested?
- Added new tests

## Did this PR include necessary documentation updates?

- Yes, I am adding a new API. I am using the [current SNAPSHOT version number](https://github.com/apache/sedona/blob/master/pom.xml#L29) in since `vX.Y.Z` format.

### CODE REVIEW

File: common/src/test/java/org/apache/sedona/common/raster/RasterBandAccessorsTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        double[] values = new double[width * height];
+        for (int i = 0; i < values.length; i++) {
+            values[i] = i + 1;
+        }
+        assertNull(RasterBandAccessors.getBandNoDataValue(emptyRaster, 1));
+    }
+
+    @Test
+    public void testBandNoDataValueIllegalBand() throws FactoryException, IOException {
+        GridCoverage2D raster = rasterFromGeoTiff(resourceFolder + "raster/raster_with_no_data/test5.tiff");
```
Comment (jiayuasu) [inline common/src/test/java/org/apache/sedona/common/raster/RasterBandAccessorsTest.java:72]: `test5` has a file extension `TIFF` but you specified `tiff` here. Please change the file extension of the original image to `tiff`.

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: common/src/test/java/org/apache/sedona/common/raster/RasterBandAccessorsTest.java (hunk 1 above)
Comment (iGN5117) [reply common/src/test/java/org/apache/sedona/common/raster/RasterBandAccessorsTest.java:72]: That didn't seem to matter for my local tests. However, I've renamed the file extension to tiff and pushed now

**Review by jiayuasu — CHANGES_REQUESTED** [review]
Please add documentation to this PR.

**Review by jiayuasu — APPROVED** [review]

File: sql/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/raster/RasterBandAccessors.scala [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package org.apache.spark.sql.sedona_sql.expressions.raster
```
Comment (jiayuasu) [inline sql/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/raster/RasterBandAccessors.scala:1]: This file must have a Apache header.

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: sql/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/raster/RasterBandAccessors.scala (hunk 2 above)
Comment (iGN5117) [reply sql/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/raster/RasterBandAccessors.scala:20]: Sorry about that, pushed

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**